source/common/router/upstream_request.cc

```diff
@@ -343,8 +342,11 @@ void UpstreamRequest::onPoolReady(Http::RequestEncoder& request_encoder,
     create_per_try_timeout_on_request_complete_ = true;
   }
 
-  conn_pool_stream_handle_ = nullptr;
-  setRequestEncoder(request_encoder);
+  // Now that there is an encoder, have the connection manager inform the manager when the
```

Comment:
this comment seems a bit out of place now that we don't have a `request_encoder` here anymore